### PR TITLE
Split up `indexing_slicing` ui test

### LINKED ISSUE #2038: Tracking Issue: Split up our UI-tests into smaller parts
Changing files like `tests/ui/methods.rs` causes a lot of diff noise. I think the ui output of every file should be kept below 100 lines preferrably and below 200 as a hard limit.

The files would best be split up to only check one or two lints

Current status:

```
$ wc -l tests/ui/*.stderr | sort -nr | head -n 15
   272 tests/ui/cognitive_complexity.stderr
   260 tests/ui/collapsible_if.stderr
   259 tests/ui/non_copy_const.stderr
   258 tests/ui/missing-doc.stderr
   247 tests/ui/if_same_then_else.stderr
   246 tests/ui/transmute.stderr
   245 tests/ui/match_same_arms.stderr
   224 tests/ui/needless_range_loop.stderr
   220 tests/ui/drop_forget_ref.stderr
   214 tests/ui/use_self.stderr
   214 tests/ui/eq_op.stderr
   213 tests/ui/indexing_slicing.stderr
   210 tests/ui/option_map_unit_fn.stderr
   203 tests/ui/booleans.stderr
```

### PR DESCRIPTION
**Pull request by JohnTitor**
Closes #2038

Now all the stderrs are less than 200 lines :tada: 

changelog: none

### CODE REVIEW

**Comment by flip1995** [comment]
I think #2038 can be closed with this. I don't think we should set a soft limit (or how we should enforce). cc @phansch

File: tests/ui/indexing_slicing_index.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    let v = vec![0; 5];
+    v[0];
+    v[10];
+    v[1 << 3];
+
+    //
+    // Continue tests at end function to minimize the changes to this file's corresponding stderr.
+    //
+
```
Comment (flip1995) [inline tests/ui/indexing_slicing_index.rs]:
```suggestion
```

This comment can be removed, I think.

Comment (JohnTitor) [reply tests/ui/indexing_slicing_index.rs]: Removed via https://github.com/rust-lang/rust-clippy/pull/5130/commits/49934e70a28d599d4b46d6885c906c5541472b2d.

**Comment by phansch** [comment]
Thanks a lot for pushing this through @JohnTitor :tada:

**Pull request merged**